lib/matplotlib/testing/__init__.py

```diff
@@ -164,3 +166,21 @@ def setup():
     rcdefaults()  # Start with all defaults
 
     set_font_settings_for_testing()
+
+
+@contextmanager
+def closed_tempfile(suffix='', text=None):
+    """
+    Context manager which yields the path to a closed temporary file with the
```

Comment:
Does publicly available mean should be used outside of matplotlib, or is the recommended option for other parts of matplotlib? I don't see much use outside matplotlib, unless you were already using matplotlib's testing utils.